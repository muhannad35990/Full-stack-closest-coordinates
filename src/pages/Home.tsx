import React, { useState } from "react";
import axios from "axios";
import Card from "../components/Card";
import styled from "styled-components";
import { WITHIN_ENDPOINT } from "../config/endpoints";
const Home = () => {
  const [km, setKm] = useState("");
  const [pointsWithin, setPointsWithin] = useState([]);
  const getPartnerswithin = (e: any) => {
    e.preventDefault();
    axios
      .get(`${WITHIN_ENDPOINT}/${km}`)
      .then((response: any) => {
        console.log(response.data.data.pointsWithin);

        setPointsWithin(response.data.data.pointsWithin);
      })
      .catch((err) => {
        console.log(err);
      });
  };
  return (
    <form method="get" onSubmit={getPartnerswithin}>
      <StyledInputContainer>
        <StyledInputSearch
          type="text"
          placeholder="Search by distance in km"
          onChange={(e) => setKm(e.target.value)}
          value={km}
        />
        <StyledButton type="submit">Search</StyledButton>
      </StyledInputContainer>

      <CardsGrid>
        {pointsWithin.map((point: any) => (
          <Card
            organization={point.organization}
            customerLocations={point.customerLocations}
            services={point.services}
            willWorkRemotely={point.willWorkRemotely}
            offices={point.offices}
          />
        ))}
      </CardsGrid>
    </form>
  );
};
const StyledInputContainer = styled.div`
  display: flex;

  align-items: center;
  justify-content: center;
`;
const StyledButton = styled.button`
  padding: 1.05rem 2rem;
  outline: none;
  border: none;
  background-color: #2998ff;
  color: white;
  border-radius: 0 10rem 10rem 0;
  &:hover {
    cursor: pointer;
  }
`;
const StyledInputSearch = styled.input`
  border: 1px solid #2998ff;
  outline: none;
  justify-self: center;
  padding: 1rem 2rem;
  width: 30vw;
  transition: all 0.2s;
  border-radius: 10rem 0 0 10rem;
  &:focus {
    border: 1px solid #2998ff;
    width: 32vw;
  }
`;
const CardsGrid = styled.div`
  margin-top: 3rem;
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  overflow: hidden;
`;
export default Home;
