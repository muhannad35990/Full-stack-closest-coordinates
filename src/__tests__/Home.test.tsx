import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Home from "../pages/Home";

describe("Home", () => {
  it("Home renders without crashing", () => {
    const wrapper = shallow(<Home />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("test search button", () => {
    const wrapper = shallow(<Home />);
    expect(wrapper.find("#searchBtn").length).toBe(1);
    expect(wrapper.find("#searchBtn").text().includes("Search")).toBe(true);
  });
  it("test search input", () => {
    const wrapper = shallow(<Home />);
    expect(wrapper.find("#SearchInput").length).toBe(1);
    expect(wrapper.find("#SearchInput").at(0).props().placeholder).toEqual(
      "Search by distance in km"
    );
  });
  describe("Form submit check", () => {
    it("changes submitted state", () => {
      const fakeEvent = { preventDefault: () => console.log("preventDefault") };
      const wrapper = shallow(<Home />);
      expect(wrapper.find("form").length).toEqual(1);
      wrapper.find("form").simulate("submit", fakeEvent);
    });
  });
});
