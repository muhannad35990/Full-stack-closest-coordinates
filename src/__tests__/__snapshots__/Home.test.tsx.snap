// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Home renders without crashing 1`] = `
<form
  method="get"
  onSubmit={[Function]}
>
  <styled.div>
    <styled.input
      id="SearchInput"
      onChange={[Function]}
      placeholder="Search by distance in km"
      type="text"
      value=""
    />
    <styled.button
      id="searchBtn"
      type="submit"
    >
      Search
    </styled.button>
  </styled.div>
  <styled.div />
</form>
`;
